app/api/common.v2/database/MongoDataSource.ts

```diff
@@ -1,28 +1,132 @@
-import { Db, Document } from 'mongodb';
+import { Collection, Db, Document } from 'mongodb';
 import { BulkWriteStream } from './BulkWriteStream';
 import { MongoTransactionManager } from './MongoTransactionManager';
 
+type MaxAmountOfParameters<F extends (...args: any) => any> = Required<Parameters<F>>['length'];
+
+/**
+ * Map of collection members to position of its options argument.
+ * Position is based on 1.
+ */
+type MethodsOptionsArgPosition = {
+  [method in keyof Collection]: Collection[method] extends (...args: any) => any
+    ? MaxAmountOfParameters<Collection[method]>
+    : null;
+};
+
 export abstract class MongoDataSource<CollectionSchema extends Document = any> {
   protected db: Db;
 
   protected abstract collectionName: string;
 
   private transactionManager: MongoTransactionManager;
 
+  private collectionProxy?: Collection<CollectionSchema>;
+
   constructor(db: Db, transactionManager: MongoTransactionManager) {
     this.db = db;
     this.transactionManager = transactionManager;
   }
 
+  static scopedMethods: MethodsOptionsArgPosition = {
```

Comment:
It's in fact a lot of work. I started it, and then dropped it because I wasn't sure it was valuable use of time.
The working assumption here is that the interface in mongo client is uniform, so I tested one or two methods, and should be good enough. But I agree it may seem a bit of a weak test strategy.